lib/notify_delivery_method.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require "notifications/client"
+
+class NotifyDeliveryMethod
+  def initialize(mail); end
+
+  def deliver!(mail)
+    client.send_email(payload(mail))
+  end
+
+private
+
+  def client
+    @client ||= Notifications::Client.new(notify_api_key)
+  end
+
+  def notify_api_key
+    notify.fetch(:api_key)
+  end
+
+  def notify
+    @notify ||= notify_environment_config.symbolize_keys.freeze
+  end
+
+  def notify_config_path
+    Rails.root.join("config", "notify.yml")
+  end
+
+  def notify_environment_config
+    YAML.safe_load(ERB.new(File.read(notify_config_path)).result).fetch(Rails.env)
```

Comment:
Might be worth breaking this up into multiple lines?

Could even be worth moving this and notify_config_path into the same method since nothing else calls them and they're short.